api/v1alpha1/cdk8sappproxy_webook.go

```diff
@@ -30,74 +34,104 @@ import (
+func (*cdk8sAppProxyWebhook) Default(_ context.Context, objRaw runtime.Object) error {
+	obj, ok := objRaw.(*Cdk8sAppProxy)
+	if !ok {
+		return apierrors.NewBadRequest(fmt.Sprintf("expected a Cdk8sAppProxy but got a %T", objRaw))
 	}
+	cdk8sappproxylog.Info("default", "name", obj.Name)
 
-	// Set the default path if not specified
-	if c.Spec.GitRepository != nil && c.Spec.GitRepository.Path == "" {
-		c.Spec.GitRepository.Path = "."
+  if obj.Spec.GitRepository.Reference == "" {
+		obj.Spec.GitRepository.Reference = "main"
 	}
+
+	return nil
 }
 
+// TODO(user): change verbs to "verbs=create;update;delete" if you want to enable deletion validation.
 // +kubebuilder:webhook:path=/validate-addons-cluster-x-k8s-io-v1alpha1-cdk8sappproxy,mutating=false,failurePolicy=fail,sideEffects=None,groups=addons.cluster.x-k8s.io,resources=cdk8sappproxies,verbs=create;update,versions=v1alpha1,name=cdk8sappproxy.kb.io,admissionReviewVersions=v1
 
-var _ webhook.Validator = &Cdk8sAppProxy{}
+// ValidateCreate implements webhook.Validator so a webhook will be registered for the type.
+func (*cdk8sAppProxyWebhook) ValidateCreate(_ context.Context, objRaw runtime.Object) (admission.Warnings, error) {
+	var allErrs field.ErrorList
 
-// ValidateCreate implements webhook.CustomValidator so a webhook will be registered for the type.
-func (c *Cdk8sAppProxy) ValidateCreate() (admission.Warnings, error) {
-	cdk8sappproxylog.Info("validate create", "name", c.Name)
-
-	return c.validateCdk8sAppProxy()
-}
+  newObj, ok := objRaw.(*Cdk8sAppProxy)
+	if !ok {
+		return nil, apierrors.NewBadRequest(fmt.Sprintf("expected a Cdk8sAppProxy but got a %T", objRaw))
+	}
 
-// ValidateUpdate implement webhook.CustomValidator so a webhook will be registered for the type.
-func (c *Cdk8sAppProxy) ValidateUpdate(oldRaw runtime.Object) (admission.Warnings, error) {
-	cdk8sappproxylog.Info("validate update", "name", c.Name)
+	cdk8sappproxylog.Info("validate create", "name", newObj.Name)
 
-	return c.validateCdk8sAppProxy()
-}
+	if newObj.Spec.GitRepository.URL == "" {
+    allErrs = append(allErrs,
+			field.Invalid(field.NewPath("spec", "GitRepository", "URL"), 
+				newObj.Spec.GitRepository.URL, "GitRepository.URL must be specified"))
+	}
 
-// ValidateDelete implements webhook.CustomValidator so a webhook will be registered for the type.
-func (c *Cdk8sAppProxy) ValidateDelete() (admission.Warnings, error) {
-	cdk8sappproxylog.Info("validate delete", "name", c.Name)
+	if len(allErrs) > 0 {
+		return nil, apierrors.NewInvalid(GroupVersion.WithKind("Cdk8sAppProxy").GroupKind(), newObj.Name, allErrs)
+	}
 
-	// No validation needed for delete
 	return nil, nil
 }
 
-func (c *Cdk8sAppProxy) validateCdk8sAppProxy() (admission.Warnings, error) {
-	var allErrs []error
+// ValidateUpdate implements webhook.Validator so a webhook will be registered for the type.
+func (*cdk8sAppProxyWebhook) ValidateUpdate(_ context.Context, oldRaw, newRaw runtime.Object) (admission.Warnings, error) {
```

Comment:
Needed to implement the Update validator, to satisfy the Interface.
The update validation logic is not the validation, we ultimately want, since updating the repoURL on an existing CAP resource should be allow - but i just need something to validate against..